browser/mapping.go

```diff
@@ -239,22 +243,56 @@ func mapFrame(vu k6modules.VU, f api.Frame) mapping {
 			return rt.ToValue(eh).ToObject(rt)
 		},
 		"getAttribute": f.GetAttribute,
-		"goto":         f.Goto,
-		"hover":        f.Hover,
-		"innerHTML":    f.InnerHTML,
-		"innerText":    f.InnerText,
-		"inputValue":   f.InputValue,
-		"isChecked":    f.IsChecked,
-		"isDetached":   f.IsDetached,
-		"isDisabled":   f.IsDisabled,
-		"isEditable":   f.IsEditable,
-		"isEnabled":    f.IsEnabled,
-		"isHidden":     f.IsHidden,
-		"isVisible":    f.IsVisible,
-		"iD":           f.ID,
-		"loaderID":     f.LoaderID,
-		"locator":      f.Locator,
-		"name":         f.Name,
+		// 👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀👀
+		//
+		// Psst, Ankur! Look here! 😄
+		//
+		// I hope you can now see why we need to pass VU in the context.
+		//
+		// I already shared this code in detail in
+		// https://github.com/grafana/xk6-browser/pull/702. I guess it
+		// makes it easier for you to see in the code, so be it :)
+		//
+		// The following code moves the promise handling to the mapping
+		// layer.
+		//
+		// Benefits:
+		// 1- This allows us to map the Response.
+		// 2- It's clearer, more manageable, and less error-prone
+		//    to have all the promise handling in one place
+		//    (in the mapping layer).
+		// 3- Tests are getting clearer (free from the promise
+		//    handling). See TestBrowserContextOptionsExtraHTTPHeaders
+		//    in browser_context_options_test.go.
+		//
+		//
+		"goto": func(url string, opts goja.Value) *goja.Promise {
+			return k6ext.Promise(ctx, func() (any, error) {
+				resp, err := f.Goto(url, opts)
+				if err != nil {
+					return nil, err //nolint:wrapcheck
+				}
+
+				return mapResponse(vu, resp), nil
+			})
```

Comment:
Handling the promise in the mapping layer. `ctx` delivers `VU` to the `k6ext.Promise`.